src/mixins/search-mixin.ts

```diff
@@ -112,15 +112,20 @@ export class SearchMixin extends Mixins(CommonMixin) {
   }
 
   set entity_type_cd (type: EntityTypes) {
-    // console.log('*** set entity_type_cd =', type)
+    // special case for amalgamation -- in case of changing entity type after changing xpro location
+    if (this.isAmalgamation) {
+      this.setSearchJurisdiction(null)
+      this.setLocation(null)
+      this.setJurisdictionCd(null)
+    }
```

Comment:
This catches two cases:
- user clicks on a short-list item (esp. after selecting a xpro entity and a xpro location)
- user clicks on "View all business types" (esp. after selecting a xpro entity and a xpro